plugins/woocommerce-blocks/assets/js/blocks/collection-filters/inner-blocks/stock-filter/frontend.ts

```diff
@@ -20,13 +21,27 @@ const getUrl = ( activeFilters: string ) => {
 
 store( 'woocommerce/collection-stock-filter', {
 	actions: {
-		// "on select" handler passed to the dropdown component.
-		navigate: () => {
-			const context = getContext< DropdownContext >(
+		onCheckboxChange: () => {
+			const checkboxContext = getContext< CheckboxListContext >(
+				'woocommerce/interactivity-checkbox-list'
+			);
+
+			const filters = checkboxContext.items
+				.filter( ( item ) => {
+					return item.checked;
+				} )
+				.map( ( item ) => {
+					return item.value;
+				} );
+
+			navigate( getUrl( filters.join( ',' ) ) );
+		},
+		onDropdownChange: () => {
```

Comment:
I think we can apply this naming convention to other block as well. It doesn't have to be implemented in this PR, but can be a part of #43167

Reply:
@dinhtungdu Yes good idea. I'll add that to the task list for that issue.